+++ /a/lib/dhcpcd/dhcpcd-hooks/10-wpa_supplicant      2019-07-19 21:31:41.130053542 +0100
+++ /b/lib/dhcpcd/dhcpcd-hooks/10-wpa_supplicant      2019-07-19 21:58:03.470829257 +0100
@@ -20,6 +20,9 @@

 wpa_supplicant_ctrldir()
 {
+        if ([ -f /etc/wpa_devices ] && ! grep -q  "$interface" /etc/wpa_devices); then
+            exit 0
+        fi
        dir=$(key_get_value "[[:space:]]*ctrl_interface=" \
                "$wpa_supplicant_conf")
        dir=$(trim "$dir")
@@ -36,6 +39,9 @@

 wpa_supplicant_start()
 {
+        if ([ -f /etc/wpa_devices ] && ! grep -q  "$interface" /etc/wpa_devices); then
+            exit 0
+        fi
        # If the carrier is up, don't bother checking anything
        [ "$ifcarrier" = "up" ] && return 0

@@ -70,6 +76,9 @@

 wpa_supplicant_reconfigure()
 {
+        if ([ -f /etc/wpa_devices ] && ! grep -q  "$interface" /etc/wpa_devices); then
+            exit 0
+        fi
        dir=$(wpa_supplicant_ctrldir)
        [ -z "$dir" ] && return 1
        if ! wpa_cli -p "$dir" -i "$interface" status >/dev/null 2>&1; then
@@ -88,6 +97,9 @@

 wpa_supplicant_stop()
 {
+        if ([ -f /etc/wpa_devices ] && ! grep -q  "$interface" /etc/wpa_devices); then
+            exit 0
+        fi
        dir=$(wpa_supplicant_ctrldir)
        [ -z "$dir" ] && return 1
        wpa_cli -p "$dir" -i "$interface" status >/dev/null 2>&1 || return 0
@@ -103,7 +115,8 @@

 if [ "$ifwireless" = "1" ] && \
     type wpa_supplicant >/dev/null 2>&1 && \
-    type wpa_cli >/dev/null 2>&1
+    type wpa_cli >/dev/null 2>&1 &&
+    ([ -f /etc/wpa_devices ] && grep -q  "$interface" /etc/wpa_devices)
 then

        case "$reason" in
